src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/Api/ReleasesController.cs

```diff
@@ -174,12 +174,39 @@ public async Task<ActionResult<FileInfo>> UpdateChartFileAsync(Guid releaseId, G
         [RequestSizeLimit(int.MaxValue)]
         [RequestFormLimits(ValueLengthLimit = int.MaxValue, MultipartBodyLengthLimit = int.MaxValue)]
         public async Task<ActionResult<DataFileInfo>> AddDataFilesAsync(Guid releaseId,
-            [FromQuery(Name = "name"), Required] string name, IFormFile file, IFormFile metaFile)
+            [FromQuery(Name = "name"), Required] string subjectName, IFormFile file, IFormFile metaFile)
         {
             var user = await _userManager.GetUserAsync(User);
 
             return await _releaseFilesService
-                .UploadDataFilesAsync(releaseId, file, metaFile, name, user.Email)
+                .UploadDataFilesAsync(
+                    releaseId: releaseId,
+                    dataFile: file,
+                    metaFile: metaFile,
+                    userName: user.Email,
+                    subjectName: subjectName)
+                .HandleFailuresOrOk();
+        }
+
+        [HttpPost("release/{releaseId}/data/replacing/{replacingId}")]
```

Comment:
Rather than have a completely new endpoint, I would suggest just adding this replacement id as a query parameter on top of the existing endpoints.

Reply:
Thanks I've done this now. I originally did it like this because subject id and replacement id are mutually exclusive so I thought it is a bit clearer.